src/main/java/com/greedobank/cards/service/CardService.java

```diff
@@ -0,0 +1,90 @@
+package com.greedobank.cards.service;
+
+import com.greedobank.cards.dao.CardDAO;
+import com.greedobank.cards.dto.CardCreationDTO;
+import com.greedobank.cards.dto.CardDTO;
+import com.greedobank.cards.dto.CardResetPinDTO;
+import com.greedobank.cards.exception.NotFoundException;
+import com.greedobank.cards.mapper.CardMapper;
+import com.greedobank.cards.model.Card;
+import com.greedobank.cards.utils.ResponseMessages;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.OffsetDateTime;
+import java.util.Random;
+
+@Service
+@RequiredArgsConstructor
+public class CardService {
+    private final CardDAO cardDAO;
+    private final CardMapper cardMapper;
+    private final ValidationCardService validationCardService;
+
+    private static final int CARD_NUMBER_LENGTH = 16;
+    private static final int PIN_CODE_LENGTH = 4;
+    private static final int CVV_LENGTH = 3;
+    private static final int DIGIT_UPPER_LIMIT = 10;
+    private static final Random RNG = new Random(System.currentTimeMillis());
+
+    public CardDTO create(CardCreationDTO cardCreationDTO) {
+        Card card = cardMapper.toCard(cardCreationDTO);
+        card.setNumber(generate(ResponseMessages.BANK_BIN, CARD_NUMBER_LENGTH));
```

Comment:
why this is not extracted to a mapper and we do mapping in the service layer directly?

Reply:
Done